plugin/src/main/java/org/autorefactor/refactoring/rules/WakeLockRefactoring.java

```diff
@@ -0,0 +1,208 @@
+            final ASTBuilder b = this.ctx.getASTBuilder();
+            TypeDeclaration typeDeclaration = (TypeDeclaration) ASTNodes.getParent(node, ASTNode.TYPE_DECLARATION);
+            ReleasePresenceChecker releasePresenceChecker = new ReleasePresenceChecker();
+            typeDeclaration.accept(releasePresenceChecker);
+            if (!releasePresenceChecker.releasePresent) {
+                Statement releaseNode = createWakelockReleaseNode(node);
+                MethodDeclaration onPauseMethod = findMethodOfType("onPause", typeDeclaration);
+                if (onPauseMethod != null && node.getParent().getNodeType() == ASTNode.EXPRESSION_STATEMENT) {
+                    r.insertAt(releaseNode,
+                            onPauseMethod.getBody().statements().size(),
+                            Block.STATEMENTS_PROPERTY,
+                            onPauseMethod.getBody());
+                    return DO_NOT_VISIT_SUBTREE;
+                } else {
+                    MethodDeclaration onPauseDeclaration = createOnPauseDeclaration();
+                    /* TODO @JnRouvignac, instead of using insertAt, calling
+                     * typeDeclaration.bodyDeclarations().add(onPauseDeclaration);
+                     * would be more intuitive.
+                     */
+                    r.insertAt(
+                            onPauseDeclaration,
+                            typeDeclaration.bodyDeclarations().size(),
+                            typeDeclaration.getBodyDeclarationsProperty(),
+                            typeDeclaration);
+                    return DO_NOT_VISIT_SUBTREE;
+                }
+            }
+        }
+        return VISIT_SUBTREE;
+    }
+
+    private Statement createWakelockReleaseNode(MethodInvocation methodInvocation) {
+        final ASTBuilder b = this.ctx.getASTBuilder();
+        IfStatement ifStatement = b.getAST().newIfStatement();
+        // test-clause
+        PrefixExpression prefixExpression = b.getAST().newPrefixExpression();
+        prefixExpression.setOperator(PrefixExpression.Operator.NOT);
+        MethodInvocation isHeldInvocation = b.getAST().newMethodInvocation();
+        isHeldInvocation.setName(b.simpleName("isHeld"));
+        isHeldInvocation.setExpression(b.copyExpression(methodInvocation));
+        prefixExpression.setOperand(isHeldInvocation);
+        ifStatement.setExpression(prefixExpression);
+        // then
+        MethodInvocation releaseInvocation = b.getAST().newMethodInvocation();
+        releaseInvocation.setName(b.simpleName("release"));
+        releaseInvocation.setExpression(b.copyExpression(methodInvocation));
+        ExpressionStatement releaseExpressionStatement = b.getAST().newExpressionStatement(releaseInvocation);
+        ifStatement.setThenStatement(b.block(releaseExpressionStatement));
+        return ifStatement;
+    }
+
+    private MethodDeclaration createOnPauseDeclaration() {
```

Comment:
And `ASTBuilder.annotation()` should use `MarkerAnnotation` instead of `NormalAnnotation`.